addons/l10n_in/views/report_invoice.xml

```diff
@@ -64,4 +68,10 @@
         </xpath>
     </template>
 
+    <template id="place_of_supply">
+        <t t-if="o.company_id.account_fiscal_country_id.code == 'IN' and o.l10n_in_state_id"
+            class="mt16"> Place of supply: <span t-esc="o.l10n_in_state_id.name" />
+        </t>
```

Comment:
```suggestion
        <div t-if="o.country_code == 'IN' and o.l10n_in_state_id">
            Place of supply: <span t-out="o.l10n_in_state_id.name" />
        </div>
```


Field `country_code` is a related field to `company_id.account_fiscal_country_id.code`
https://github.com/odoo/odoo/blob/16e8de01b14ddac69cd706f8f64e762406769542/addons/account/models/account_move.py#L246


`t-esc` is deprecated. Please use `t-out`
https://github.com/odoo/odoo/blob/16e8de01b14ddac69cd706f8f64e762406769542/odoo/addons/base/models/ir_qweb.py#L297-L299

In your version the `class` was on a `t` tag and thus did not make a difference (tested locally). So I removed it in my suggestion. But if the "Place of supply" should be separated by white space from the address you need to re-add it. Please check functionally.